include/ws_functions/pwg.users.php

```diff
@@ -622,4 +622,67 @@ function ws_users_setInfo($params, &$service)
     ));
 }
 
+/**
+ * API method
+ * Returns the favorite images of the current user
+ * @param mixed[] $params
+ */
+function ws_getFavorites($params, &$service)
+{
+  global $conf, $user;
+  if (is_a_guest())
+  {
+    return false;
+  }
+
+  $search_user = $user['id'];
+  if (is_admin() && is_numeric($params['user_id']))
+  {
+  	$search_user = $params['user_id'];
+  }
+
+  check_user_favorites();
+  $order_by = ws_std_image_sql_order($params, 'i.');
+  $order_by = empty($order_by) ? $conf['order_by'] : 'ORDER BY '.$order_by;
+  $query = 'SELECT i.*
+    FROM '.FAVORITES_TABLE.'
+    INNER JOIN '.IMAGES_TABLE.' i ON image_id = i.id
+    WHERE user_id = '.$search_user.'
+    '.get_sql_condition_FandF( array('visible_images' => 'id'), 'AND').'
```

Comment:
get_sql_condition_FandF relies on the current $user, not on $search_user. You can't accurately use this function if you $search_user != $user['id'].